modules/javafx.graphics/src/main/java/com/sun/glass/ui/WindowControlsOverlay.java

```diff
@@ -0,0 +1,539 @@
+                StyleConverter.getEnumConverter(HorizontalDirection.class),
+                HorizontalDirection.RIGHT) {
+            @Override
+            public boolean isSettable(WindowControlsOverlay overlay) {
+                return true;
+            }
+
+            @Override
+            public StyleableProperty<HorizontalDirection> getStyleableProperty(WindowControlsOverlay overlay) {
+                return overlay.buttonPlacement;
+            }
+        };
+
+    private static final CssMetaData<WindowControlsOverlay, Boolean> ALLOW_RTL_METADATA =
+        new CssMetaData<>("-fx-allow-rtl", StyleConverter.getBooleanConverter(), true) {
+            @Override
+            public boolean isSettable(WindowControlsOverlay overlay) {
+                return true;
+            }
+
+            @Override
+            public StyleableProperty<Boolean> getStyleableProperty(WindowControlsOverlay overlay) {
+                return overlay.allowRtl;
+            }
+        };
+
+    private static final List<CssMetaData<?, ?>> METADATA =
+        Stream.concat(getClassCssMetaData().stream(),
+        Stream.of(BUTTON_PLACEMENT_METADATA, ALLOW_RTL_METADATA)).toList();
+
+    private static final PseudoClass HOVER_PSEUDOCLASS = PseudoClass.getPseudoClass("hover");
+    private static final PseudoClass PRESSED_PSEUDOCLASS = PseudoClass.getPseudoClass("pressed");
+    private static final PseudoClass ACTIVE_PSEUDOCLASS = PseudoClass.getPseudoClass("active");
+    private static final String DARK_STYLE_CLASS = "dark";
+    private static final String RESTORE_STYLE_CLASS = "restore";
+
+    /**
+     * The metrics (placement and size) of the window buttons.
+     */
+    private final ObjectProperty<WindowOverlayMetrics> metrics = new SimpleObjectProperty<>(
+        this, "metrics", new WindowOverlayMetrics(HorizontalDirection.RIGHT, new Dimension2D(0, 0), 0));
+
+    /**
+     * Specifies the placement of the window buttons on the left or the right side of the window.
+     * <p>
+     * This property corresponds to the {@code -fx-button-placement} CSS property.
+     */
+    private final StyleableObjectProperty<HorizontalDirection> buttonPlacement =
+        new SimpleStyleableObjectProperty<>(
+                BUTTON_PLACEMENT_METADATA, this, "buttonPlacement", HorizontalDirection.RIGHT) {
+            @Override
+            protected void invalidated() {
+                requestLayout();
+            }
+        };
+
+    /**
+     * Specifies whether the minimize/maximize/close buttons support right-to-left orientations.
+     * <p>
+     * If this property is {@code true} and the effective node orientation is right-to-left, the
+     * window buttons are mirrored to the other side of the window.
+     * <p>
+     * This property corresponds to the {@code -fx-allow-rtl} CSS property.
+     */
+    private final StyleableBooleanProperty allowRtl =
+        new SimpleStyleableBooleanProperty(ALLOW_RTL_METADATA, this, "allowRtl", true) {
```

Comment:
The purpose is to allow JavaFX developers (that is us, not users of JavaFX) to define the entire appearance of `WindowControlsOverlay` with a CSS file. This should make it easier to maintain it, since we won't have to keep looking in two places to change the appearance.

Reply:
I don't understand: why the orientation of EXTENDED window be different from that of the platform (unless explicitly configured with Scene.nodeOrientation)?

Reply:
When `-fx-allow-rtl` is set and the node orientation is RTL, the window buttons move to the other side of the window. This is a bit of a stylistic decision that may depend on OS and window manager. Maybe some window managers don't do that, and instead always keep the window buttons on the same side. This if kind of future-proofing the `WindowControlsOverlay` control.
Let's say that in the future, a popular Linux window manager emerges that has a very distinct look and always keeps the window buttons on the same side. Then supporting this window manager will be as easy as providing an appropriate CSS file for its look-and-feel, all without changing the `WindowControlsOverlay` control.